vpr/src/base/vpr_constraints_reader.cpp

```diff
@@ -31,10 +31,13 @@ void load_vpr_constraints_file(const char* read_vpr_constraints_name) {
             read_vpr_constraints_name);
     }
 
-    VprConstraints constraints = reader.constraints_;
-    int num_parts = reader.num_partitions_;
+    //Update the floorplanning constraints in the floorplanning constraints context
+    auto& floorplanning_ctx = g_vpr_ctx.mutable_floorplanning();
+    floorplanning_ctx.constraints = reader.constraints_;
+
+    VprConstraints ctx_constraints = floorplanning_ctx.constraints;
 
     if (getEchoEnabled() && isEchoFileEnabled(E_ECHO_VPR_CONSTRAINTS)) {
-        echo_constraints(getEchoFileName(E_ECHO_VPR_CONSTRAINTS), constraints, num_parts);
+        echo_constraints(getEchoFileName(E_ECHO_VPR_CONSTRAINTS), ctx_constraints, ctx_constraints.get_num_partitions());
```

Comment:
don't need to pass third argument, since you're passing ctx_constraints, can just call ctx_constraints.get_num_partitions() inside echo_constraints - also pass ctx_constraints by const reference as a good convention